src/pages/index.js

```diff
@@ -202,3 +193,148 @@ function closeModalOnEsc(evt) {
 }
 
 enableValidation(settings);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "1c9b6b75-fdbb-45e6-b261-c4a3ce5c7c23",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    profileName.textContent = userData.name;
+    profileDescription.textContent = userData.about;
+    profileAvatar.src = userData.avatar;
+    userId = userData._id;
+
+    return api.getInitialCards();
+  })
+  .then((initialCards) => {
+    initialCards.forEach((card) => {
+      const cardElement = getCardElement(card);
+      galleryList.prepend(cardElement);
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+// -----Submit New Post to Gallery-----
+
+function handleNewPostFormSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = {
+    name: inputCaption.value,
+    link: inputImageLink.value,
+  };
+  api
+    .addCard(inputValues)
+    .then((newCardData) => {
+      const newCard = getCardElement(newCardData);
+      return galleryList.prepend(newCard);
+    })
+    .then(() => {
+      closeModal(newPostModal);
+      newPostForm.reset();
+      disableButton(newPostSubmitButton, settings);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+newPostForm.addEventListener("submit", handleNewPostFormSubmit);
+
+// -----Submit changes from Form and Close Edit Avatar Modal-----
+
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const originalSubmitAvatarButton = submitAvatarButton.textContent;
+  const errorElement = document.querySelector("#avatar-image-input-error");
+  errorElement.textContent = "";
+  submitAvatarButton.textContent = "Saving...";
+  const newAvatarUrl = inputAvatarLink.value;
+
+  if (!newAvatarUrl) {
+    errorElement.textContent = "Please enter a valid URL";
+    submitAvatarButton.textContent = originalSubmitAvatarButton;
+    return;
+  }
+  api
+    .updateAvatar(newAvatarUrl)
+    .then((updatedUserData) => {
+      profileAvatar.src = updatedUserData.avatar;
+      submitAvatarButton.textContent = originalSubmitAvatarButton;
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.  You don't need to duplicate it in `.then` and `.catch`